src/main/java/org/example/bookstore/security/CustomUserDetailsService.java

```diff
@@ -0,0 +1,21 @@
+package org.example.bookstore.security;
+
+import lombok.RequiredArgsConstructor;
+import org.example.bookstore.repository.user.UserRepository;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.stereotype.Service;
+
+@RequiredArgsConstructor
+@Service
+public class CustomUserDetailsService implements UserDetailsService {
+    private final UserRepository userRepository;
+
+    @Override
+    public UserDetails loadUserByUsername(String email)
+            throws UsernameNotFoundException {
+        return userRepository.findByEmail(email)
+                .orElseThrow(() -> new UsernameNotFoundException(email));
```

Comment:
```suggestion
                .orElseThrow(() -> new EntityNotFoundException(email));
```

provide informative message